test/kubernetes_test.go

```diff
@@ -34,189 +36,244 @@ var dnsTestCases = []test.Case{
+
+		Extra: append([]dns.RR{
+			test.A("svc-c.test-1.svc.cluster.local.	0	IN	A	10.0.0.115"),
+		}, getSRVrecords("_c-port._UDP.*.test-1.svc.cluster.local.", "TypeA", "headless-svc", "test-1")...),
 	},
 	{
 		Qname: "*._tcp.any.test-1.svc.cluster.local.", Qtype: dns.TypeSRV,
 		Rcode: dns.RcodeSuccess,
 		Answer: []dns.RR{
-			test.SRV("_http._tcp.svc-1-a.test-1.svc.cluster.local.      303    IN    SRV 10 100 80 svc-1-a.test-1.svc.cluster.local."),
-			test.SRV("_https._tcp.svc-1-a.test-1.svc.cluster.local.      303    IN    SRV 10 100 443 svc-1-a.test-1.svc.cluster.local."),
-			test.SRV("_http._tcp.svc-1-b.test-1.svc.cluster.local.      303    IN    SRV 10 100 80 svc-1-b.test-1.svc.cluster.local."),
+			test.SRV("*._tcp.any.test-1.svc.cluster.local.      303    IN    SRV 0 33 80 svc-1-a.test-1.svc.cluster.local."),
+			test.SRV("*._tcp.any.test-1.svc.cluster.local.      303    IN    SRV 0 33 443 svc-1-a.test-1.svc.cluster.local."),
+			test.SRV("*._tcp.any.test-1.svc.cluster.local.      303    IN    SRV 0 33 80 svc-1-b.test-1.svc.cluster.local."),
+		},
+		Extra: []dns.RR{
+			test.A("svc-1-a.test-1.svc.cluster.local.	0	IN	A	10.0.0.100"),
+			test.A("svc-1-b.test-1.svc.cluster.local.	0	IN	A	10.0.0.110"),
 		},
 	},
 	{
 		Qname: "*.*.any.test-2.svc.cluster.local.", Qtype: dns.TypeSRV,
 		Rcode:  dns.RcodeNameError,
 		Answer: []dns.RR{},
+		Ns: []dns.RR{
+			test.SOA("cluster.local.	300	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1502313310 7200 1800 86400 60"),
+		},
 	},
 	{
 		Qname: "*.*.*.test-2.svc.cluster.local.", Qtype: dns.TypeSRV,
 		Rcode:  dns.RcodeNameError,
 		Answer: []dns.RR{},
+		Ns: []dns.RR{
+			test.SOA("cluster.local.	300	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1502313310 7200 1800 86400 60"),
+		},
 	},
 	{
 		Qname: "_http._tcp.*.*.svc.cluster.local.", Qtype: dns.TypeSRV,
 		Rcode: dns.RcodeSuccess,
 		Answer: []dns.RR{
-			test.SRV("_http._tcp.svc-1-a.test-1.svc.cluster.local.      303    IN    SRV 10 100 80 svc-1-a.test-1.svc.cluster.local."),
-			test.SRV("_http._tcp.svc-1-b.test-1.svc.cluster.local.      303    IN    SRV 10 100 80 svc-1-b.test-1.svc.cluster.local."),
+			test.SRV("_http._tcp.*.*.svc.cluster.local.      303    IN    SRV 0 50 80 svc-1-a.test-1.svc.cluster.local."),
+			test.SRV("_http._tcp.*.*.svc.cluster.local.      303    IN    SRV 0 50 80 svc-1-b.test-1.svc.cluster.local."),
+		},
+		Extra: []dns.RR{
+			test.A("svc-1-a.test-1.svc.cluster.local.	0	IN	A	10.0.0.100"),
+			test.A("svc-1-b.test-1.svc.cluster.local.	0	IN	A	10.0.0.110"),
 		},
 	},
 	{
 		Qname: "*.svc-1-a.test-1.svc.cluster.local.", Qtype: dns.TypeSRV,
 		Rcode:  dns.RcodeNameError,
 		Answer: []dns.RR{},
+		Ns: []dns.RR{
+			test.SOA("cluster.local.	300	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1502313310 7200 1800 86400 60"),
+		},
 	},
 	{
 		Qname: "*._not-udp-or-tcp.svc-1-a.test-1.svc.cluster.local.", Qtype: dns.TypeSRV,
 		Rcode:  dns.RcodeNameError,
```

Comment:
This Rcode is wrong; this should be dns.RcodeSucccess and the message reply should indicate a nodata reponse.